### ISSUE TITLE
feat: Integrate with delta lake

### ISSUE THREAD

**Issue by Xuanwo** [databendlabs/databend#7974]
**Summary**

Delta lake is an open-source storage framework that enables building a Lakehouse architecture with compute engines including Spark, PrestoDB, Flink, Trino, and Hive and APIs.

Allow databend read data from delta lake seems interesting. We can use [delta-rs](https://github.com/delta-io/delta-rs).

**Comment by eastfisher** [comment]
Does it mean using Databend as a compute engine for Delta, just like Spark / Presto?

**Comment by Xuanwo** [comment]
> Does it mean using Databend as a compute engine for Delta, just like Spark / Presto?

Yes, make databend-query can query over delta lake directly. Integrate `delta-rs` is not suitable for us now, we need to port `delta-rs` to opendal first.

**Comment by davidgasquez** [comment]
Hey there! Are you planning on working on this one in 2023? Didn't saw Delta @ https://github.com/datafuselabs/databend/issues/9448. :cry:

**Comment by sundy-li** [comment]
> Hey there! Are you planning on working on this one in 2023? Didn't saw Delta @ #9448. 😢

Thanks, it's added!
